tmt/steps/__init__.py

```diff
@@ -863,42 +871,55 @@ def login(context: Any, **kwargs: Any) -> None:
 
     @classmethod
     def plugins(cls, step: Step) -> List['Login']:
+        from tmt.steps.execute import ExecutePlugin
         """ Return list of login instances for given step """
         if not Login._enabled:
             return []
+        if cls._opt('test') and step.name == 'execute':
+            for p in step.phases(classes=(ExecutePlugin,)):
+                p._login_after_test = Login(step, 90)
```

Comment:
I'd propose to leave the implementation to the plugin itself. We can add a default implementation to `ExecutePlugin`, but I for one would stay away from changing the plugin's "private" attributes. This would be a clear internal API of "execute" plugins, e.g. `p.login_afte_test = True` - tell the plugin what you want it to do, not how to do it. Plugins would get a free hand to rely on default implementation or use custom code, e.g. custom class based on `Login`.

```python
_login_after_test: Optional[tmt.steps.Login] = None

@property
def login_after_test(self) -> bool:
  return self._login_after_test is not None

@login_after_test.setter
def login_after_test(self, value) -> None:
  self._login_after_test = Login(self.step, 90) if value else None
  # of plugin's SuperCoolMuchBetterLogin class derived from Login but playing nice jingle when started
```

Reply:
This should be also implemented. Please have a look.

Reply:
Thank you, LGTM!